enterprise/cmd/frontend/internal/auth/bitbucketcloudoauth/config.go

```diff
@@ -61,16 +63,16 @@ func parseConfig(logger log.Logger, cfg conftypes.SiteConfigQuerier, db database
 			continue
 		}
 
-		if _, ok := configured[provider.ServiceID]; ok {
-			problems = append(problems, conf.NewSiteProblems(fmt.Sprintf(`Cannot have more than one auth provider with url %q, only the first one will be used`, provider.ServiceID))...)
+		if existingProviders.Has(provider.CachedInfo().UniqueID()) {
+			problems = append(problems, conf.NewSiteProblems(fmt.Sprintf(`Cannot have more than one auth provider with url %q and client ID %q, only the first one will be used`, provider.ServiceID, provider.CachedInfo().ClientID))...)
```

Comment:
nit: let's also specify code host type in the message (same as with ADO)
```suggestion
			problems = append(problems, conf.NewSiteProblems(fmt.Sprintf(`Cannot have more than one auth provider for Bitbucket Cloud with url %q and client ID %q, only the first one will be used`, provider.ServiceID, provider.CachedInfo().ClientID))...)
```